src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/Rca.java

```diff
@@ -24,11 +24,28 @@
 
 public abstract class Rca<T extends ResourceFlowUnit> extends NonLeafNode<T> {
   private static final Logger LOG = LogManager.getLogger(Rca.class);
+  // the amount of RCA period this RCA needs to run before sending out a flowunit
+  protected final int rcaPeriod;
+  protected boolean alwaysCreateSummary;
+  protected int counter;
 
-  public Rca(long evaluationIntervalSeconds) {
+  public Rca(long evaluationIntervalSeconds, int rcaPeriod) {
```

Comment:
Going back we have a plan to remove the `evaluationIntervalSeconds` because we think that RCA framework should be simple. It should just run every graph node at every tick. The node decides if it wants 
- to not do any work and return a empty flow unit.
- to do some work and yet return a empty flow unit as it is not time to emit one
- to do some work and emit a flow unit.

So basically the framework runs each graph node each time and it is upto the node to do whatever it wants and return a flow unit empty or otherwise. 
So, we should try to not add the new rcaPeriod

Reply:
I would go as far to recommend that we make sure that no RCA depends on `evaluationIntervalSeconds`(by that I mean no RCA requires to be scheduled at a particular interval or depend on that.) as part of this CR and make it vestigial and then in a subsequent CR we can remove it.

Reply:
Done.